src/utils/api.js

```diff
@@ -11,7 +11,26 @@ import {
   setDoc,
 } from "firebase/firestore";
 import { getDatabase, ref, set } from "firebase/database";
+import {
+  getStorage,
+  uploadBytes,
+  getDownloadURL,
+  uploadBytesResumable,
+  ref as ref_storage,
+} from "firebase/storage";
+import { getAuth, onAuthStateChanged } from "firebase/auth";
+
 export const db = getFirestore();
+const storage = getStorage();
+const auth = getAuth();
+
+let uid;
+onAuthStateChanged(auth, (user) => {
```

Comment:
Is this just to get the current user ID? If so that info is already available from data passed down to postListing via user_id, which can then be passed to upload image.